src/dataset.py

```diff
@@ -1,24 +1,94 @@
+import csv
+
 import numpy as np
-from typing import Dict
+from typing import List, Optional, Tuple
+from loss import Information_gain
 
 
 class Dataset:
-    def __init__(self, dataset: Dict[str, np.ndarray]):
+    def __init__(self, dataset: np.array, labels_index: int):
         self._data = dataset
+        self._labels_index = labels_index
 
-    def load_from_csv(self, path) -> 'Dataset':
-        pass
+    # Idea for the loading data: creating another module which allows to generalize the format of the data imported
+    # Using for example an abstract class for the loading package.
 
-    def load_from_array(self, data: np.array) -> 'Dataset':
-        pass
+    @staticmethod
+    def load_from_csv(path: str, csv_limiter: str, label_name: str, new_line: str = '') -> 'Dataset':
 
+        with open(path, newline=new_line) as csvfile:
+            data = []
+            reader = csv.reader(csvfile, delimiter=csv_limiter)
+            for row in reader:
+                data.append(row)
 
-    def chose_best_split(self):
-        # TODO : This function contains the algo which allows us to take the split from the dataset. I have to think about the link of a condition
-        pass
+            label_index = data[0].index(label_name)
```

Comment:
I'd be careful here as I wasn't sure what label_index was about exactly.
As I understand you're getting the top row in order to get the names of each column? If so you are getting 2 things: all the feature names and the name of the label "category". Then within the label column you will have multiple labels. 
I would try to make that processing clear now so it's easier to use later on in your code.

E.g. label category might be "animal" and then the actual labels might be "cat", "dog"